### ISSUE TITLE
Support dropout and other layers != MLP

### ISSUE THREAD

**Issue by martelloni** [MusicallyEmbodiedML/MLP_XMOS#7]
Refactor creation of network so that we follow a Pytorch-style anywhere-to-anywhere network architecture. We need:
- MLP
- Reservoir
- Echo state network
- Dropout
- Batch norm
